sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py

```diff
@@ -135,12 +183,22 @@ def _start_producer(self, partition_id, send_timeout):
                 not self._producers[partition_id]
                 or cast(EventHubProducer, self._producers[partition_id]).closed
             ):
+                partition_config = self._partition_configs.get(partition_id)
                 self._producers[partition_id] = self._create_producer(
-                    partition_id=partition_id, send_timeout=send_timeout
+                    partition_id=partition_id,
+                    send_timeout=send_timeout,
+                    enable_idempotent_partitions=self._config.enable_idempotent_partitions,
+                    partition_config=partition_config
                 )
 
-    def _create_producer(self, partition_id=None, send_timeout=None):
-        # type: (Optional[str], Optional[Union[int, float]]) -> EventHubProducer
+    def _create_producer(
+        self,
+        partition_id=None,
+        send_timeout=None,
+        enable_idempotent_partitions=False,
+        partition_config=None
+    ):
+        # type: (Optional[str], Optional[Union[int, float]], bool, PartitionPublishingConfigType) -> EventHubProducer
```

Comment:
I already have the type definition:
```
PartitionPublishingConfigType = Optional[Union[PartitionPublishingConfiguration, dict]]
```

and for the bool, according to the [typing docs](https://docs.python.org/3/library/typing.html#typing.Optional), I don't think it has to be optional?
```
typing.Optional
Optional type.

Optional[X] is equivalent to Union[X, None].

Note that this is not the same concept as an optional argument, which is one that has a default. An optional argument with a default does not require the Optional qualifier on its type annotation just because it is optional. For example:

def foo(arg: int = 0) -> None:
    ...
On the other hand, if an explicit value of None is allowed, the use of Optional is appropriate, whether the argument is optional or not. For example:

def foo(arg: Optional[int] = None) -> None:
    ...
```